numpy/core/src/multiarray/methods.c

```diff
@@ -2084,7 +2098,13 @@ array_setstate(PyArrayObject *self, PyObject *args)
                 Py_DECREF(rawdata);
                 Py_RETURN_NONE;
             }
-            fa->data = PyDataMem_NEW(num);
+            /* Store the functions in case the global handler is modified */
```

Comment:
Is this comment still valid or was it for when `NULL` was special?

Reply:
This comment is invalid since the thread-locality commit. As far as I remember, it appears somewhere else too.

Reply:
The comment is almost valid, it should relate to "context" not "global". Rewording.

Reply:
fixed

Reply:
It seems that I deleted my previous reply accidentally.

Since the context/thread locality commit, actually, the `current_allocator` can't change inside the function's scope. That comment appears elsewhere too.

Reply:
this is less about the function scope and more about the lifetime of the `fa`